pkg/planner/core/casetest/rule/testdata/join_reorder_suite_out.json

```diff
@@ -1597,5 +1597,189 @@
+          "│ │ └─Selection 9990.00 cop[tikv]  not(isnull(test.t4.a))",
+          "│ │   └─TableFullScan 10000.00 cop[tikv] table:t4 keep order:false, stats:pseudo",
+          "│ └─HashJoin(Probe) 12487.50 root  inner join, equal:[eq(test.t5.a, test.t6.a)]",
+          "│   ├─TableReader(Build) 9990.00 root  data:Selection",
+          "│   │ └─Selection 9990.00 cop[tikv]  not(isnull(test.t6.a))",
+          "│   │   └─TableFullScan 10000.00 cop[tikv] table:t6 keep order:false, stats:pseudo",
+          "│   └─TableReader(Probe) 9990.00 root  data:Selection",
+          "│     └─Selection 9990.00 cop[tikv]  not(isnull(test.t5.a))",
+          "│       └─TableFullScan 10000.00 cop[tikv] table:t5 keep order:false, stats:pseudo",
+          "└─HashJoin(Probe) 15609.38 root  inner join, equal:[eq(test.t1.a, test.t2.a)]",
+          "  ├─TableReader(Build) 9990.00 root  data:Selection",
+          "  │ └─Selection 9990.00 cop[tikv]  not(isnull(test.t1.a))",
+          "  │   └─TableFullScan 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo",
+          "  └─HashJoin(Probe) 12487.50 root  inner join, equal:[eq(test.t2.a, test.t3.a)]",
+          "    ├─TableReader(Build) 9990.00 root  data:Selection",
+          "    │ └─Selection 9990.00 cop[tikv]  not(isnull(test.t3.a))",
+          "    │   └─TableFullScan 10000.00 cop[tikv] table:t3 keep order:false, stats:pseudo",
+          "    └─TableReader(Probe) 9990.00 root  data:Selection",
+          "      └─Selection 9990.00 cop[tikv]  not(isnull(test.t2.a))",
+          "        └─TableFullScan 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ],
+        "Warning": null
+      },
+      {
+        "SQL": "select /*+ leading(t1@sel_1, (t2@sel_2, t3@sel_3)) */ * from (select * from t1) t1, (select * from t2) t2, (select * from t3) t3 where t1.a=t2.a and t2.a=t3.a;",
+        "Plan": [
+          "HashJoin 15609.38 root  inner join, equal:[eq(test.t1.a, test.t2.a)]",
+          "├─TableReader(Build) 9990.00 root  data:Selection",
+          "│ └─Selection 9990.00 cop[tikv]  not(isnull(test.t1.a))",
+          "│   └─TableFullScan 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo",
+          "└─HashJoin(Probe) 12487.50 root  inner join, equal:[eq(test.t2.a, test.t3.a)]",
+          "  ├─TableReader(Build) 9990.00 root  data:Selection",
+          "  │ └─Selection 9990.00 cop[tikv]  not(isnull(test.t3.a))",
+          "  │   └─TableFullScan 10000.00 cop[tikv] table:t3 keep order:false, stats:pseudo",
+          "  └─TableReader(Probe) 9990.00 root  data:Selection",
+          "    └─Selection 9990.00 cop[tikv]  not(isnull(test.t2.a))",
+          "      └─TableFullScan 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ],
+        "Warning": [
+          "Warning 1815 There are no matching table names for (t2, t3) in optimizer hint /*+ LEADING(t1, t2, t3) */. Maybe you can use the table alias name"
```

Comment:
Another minor improvement is the warning message for the origin hint `/*+ LEADING(t1, t2, t3) */` seems different from the using hint `/*+ leading(t1@sel_1, (t2@sel_2, t3@sel_3)) */`. Is this can be improved?